add dumpling-lightning-job (#2770)
pingcap/docs-tidb-operator: en/restore-from-blob-using-job.md

---
title: Restore Backup Data from Azure Blob Storage Using TiDB Lightning

> summary: This document describes how to use TiDB Lightning to restore backup data stored in Azure Blob Storage to a TiDB cluster.

Reviewer: ```suggestion
summary: Learn how to use TiDB Lightning to restore backup data stored in Azure Blob Storage to a TiDB cluster.
```

This document describes how to use [TiDB Lightning](https://docs.pingcap.com/tidb/stable/tidb-lightning-overview/) to restore backup data from Azure Blob Storage to a TiDB cluster. TiDB Lightning is a tool for fast full data import into a TiDB cluster. This document uses the [physical import mode](https://docs.pingcap.com/tidb/stable/tidb-lightning-physical-import-mode/). The following example shows how to restore backup data from Azure Blob Storage to a TiDB cluster.

## Prepare a node pool for TiDB Lightning

You can run TiDB Lightning in an existing node pool or create a dedicated node pool. The following command example shows how to create a new node pool. Replace the variables as needed:

- `${clusterName}`: AKS cluster name
- `${resourceGroup}`: Resource group name

```shell
az aks nodepool add --name lightning \
    --cluster-name ${clusterName} \
    --resource-group ${resourceGroup} \
    --zones 1 2 3 \
    --node-count 1 \
    --labels dedicated=lightning
```

## Deploy the TiDB Lightning job

This section describes how to configure, deploy, and monitor the TiDB Lightning Job.

### Configure the TiDB Lightning job

> Below is a sample configuration file (`lightning_job.yaml`) for the TiDB Lightning Job. Replace the variables as needed:

Reviewer 2: ```suggestion
The following is a sample configuration file (`lightning_job.yaml`) for the TiDB Lightning job. Replace the variables as needed:
```

                  --sorted-kv-dir=/var/lib/sorted-kv \
                  --d=azure://external/testfolder?account-name=${accountname}&account-key=${accountkey} \
                  --config=/etc/tidb-lightning/tidb-lightning.toml \
                  --log-file="-"
          volumeMounts:
            - name: config
              mountPath: /etc/tidb-lightning
            - name: sorted-kv
              mountPath: /var/lib/sorted-kv
      volumes:
        - name: config
          configMap:
            name: ${name}
            items:
            - key: config-file
              path: tidb-lightning.toml
        - name: sorted-kv
          persistentVolumeClaim:
            claimName: ${name}-sorted-kv
      restartPolicy: Never
  backoffLimit: 0
```

### Create the TiDB Lightning job

Run the following commands to create the TiDB Lightning job:

```shell
export name=lightning
export version=v8.5.1
export namespace=tidb-cluster
export storageClassName=<your-storage-class>
export storage=250G
export accountname=<your-account-name>
export accountkey=<your-account-key>

envsubst < lightning_job.yaml | kubectl apply -f -
```

### Check the TiDB Lightning job status

> Run the following command to check the Pod status of the TiDB Lightning Job:

Reviewer 2: ```suggestion
Run the following command to check the Pod status of the TiDB Lightning job:
```

```shell
kubectl -n ${namespace} get pod ${name}
```

### View TiDB Lightning Job Logs

> Run the following command to view the logs of the TiDB Lightning Job:

Reviewer 2: ```suggestion
Run the following command to view the logs of the TiDB Lightning job:
```
